examples/xds-management-server/pkg/server/server.go

```diff
@@ -0,0 +1,63 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	discoveryservice "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	endpointservice "github.com/envoyproxy/go-control-plane/envoy/service/endpoint/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/server/v3"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+const (
+	grpcMaxConcurrentStreams = 1000000
+)
+
+// Run starts a go-control-plane, xds grpc server in a background goroutine.
+// The server is bounded by the provided context.
+func Run(
```

Comment:
I mentioned this in snapshot.go - but I'd love to see this split into a New(args...) and Run(ctx) pattern. It'll mean you will need a new struct - but I expect it will make it easier for our end users to extend in the long run.